flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/BatchShuffleReadBufferPool.java

```diff
@@ -75,6 +75,10 @@
     @GuardedBy("buffers")
     private final Queue<MemorySegment> buffers = new ArrayDeque<>();
 
+    /** The timestamp when the last buffer is recycled or allocated. */
+    @GuardedBy("buffers")
+    private long lastBufferOperationTimestamp = System.nanoTime();
```

Comment:
nit: maybe we can add a test to guard that this value is updated after requesting/recycling buffers successfully. If buffer requesting is not successful the value should be not updated.

Reply:
OK, Added the test case `testBufferOperationTimestampUpdated` and `testBufferOperationTimestampKeepSame`.

Reply:
OK, added the test case `testBufferOperationTimestampUpdated` in `BatchShuffleReadBufferPoolTest`.